picker_wrapper.go

```diff
@@ -120,6 +120,14 @@ func (bp *pickerWrapper) pick(ctx context.Context, failfast bool, opts balancer.
 			bp.mu.Unlock()
 			select {
 			case <-ctx.Done():
+				if bp.connectionError() != nil {
```

Comment:
Store `bp.connectionError` in a variable so you don't need to call it multiple times.
Also, if you call it more than once, the result may be different (when it's updated in between).

Reply:
> Store `bp.connectionError` in a variable so you don't need to call it multiple times.
> Also, if you call it more than once, the result may be different (when it's updated in between).
`the result may be different (when it's updated in between).`
I realized this problem. I checked the caller of `updateConnectionError`. I think we should return the latest error. 
I will change it to what you said.

Reply:
@menghanl I changed it.